docs/how-to-guides/data-basics/5-adding-a-delete-button.md

```diff
@@ -0,0 +1,336 @@
+
+Let's dispatch that action when `DeletePageButton` is clicked:
+
+```js
+const DeletePageButton = ({ pageId }) => {
+	const { deleteEntityRecord } = useDispatch( coreDataStore );
+	const handleDelete = () => deleteEntityRecord( 'postType', 'page', pageId );
+	return (
+		<Button variant="primary" onClick={ handleDelete }>
+			Delete
+		</Button>
+	);
+}
+```
+
+### Step 3: Add visual feedback
+
+It may take a few moments for the REST API request to finish after clicking the _Delete_ button. Let's communicate that with a `<Spinner />` component in a similar fashion as we did in the previous parts of this tutorial.
+
+We'll need the `isDeletingEntityRecord` selector for that. It is similar to the `isSavingEntityRecord` selector we've already seen in [part 3](/docs/how-to-guides/data-basics/3-building-an-edit-form.md): it returns `true` or `false` and never issues any HTTP requests:
+
+```js
+const DeletePageButton = ({ pageId }) => {
+	// ...
+	const { isDeleting } = useSelect(
+		select => ({
+			isDeleting: select( coreDataStore ).isDeletingEntityRecord( 'postType', 'page', pageId ),
+		}),
+		[ pageId ]
+	)
+	return (
+		<Button variant="primary" onClick={ handleDelete } disabled={ isDeleting }>
+			{ isDeleting ? (
+				<>
+					<Spinner />
+					Deleting...
+				</>
+			) : 'Delete' }
+		</Button>
+	);
+}
+```
+
+Here's what it looks like in action:
+
+![](https://raw.githubusercontent.com/WordPress/gutenberg/HEAD/docs/how-to-guides/data-basics/media/delete-button/deleting-in-progress.png)
+
+### Step 4: Handle errors
+
+We optimistically assumed that a *delete* operation would always succeed. Unfortunately, under the hood, it is a REST API request that can fail in many ways:
+
+* The website can be down
+* The update may be invalid
+* The page could have been deleted by someone else in the meantime
+
+To tell the user when any of these happens, we need to extract the error information using the `getLastEntityDeleteError` selector:
+
+```js
+// Replace 9 with an actual page ID
+wp.data.select( 'core' ).getLastEntityDeleteError( 'postType', 'page', 9 )
+```
+
+Here's how we can apply it in `DeletePageButton`:
+
+```js
+import { useEffect } from '@wordpress/element';
+const DeletePageButton = ({ pageId }) => {
+	// ...
+	const { error, /* ... */ } = useSelect(
+		select => ( {
+			error: select( coreDataStore ).getLastEntityDeleteError( 'postType', 'page', pageId ),
+			// ...
+		} ),
+		[pageId]
+	);
+	useEffect( () => {
+		if ( error ) {
+			alert( error );
+		}
+	}, [error] )
+
+	// ...
+}
+```
+
+Great! `DeletePageButton` is now fully aware of errors (even if `alert` is not the prettiest way to communicate them).
```

Comment:
I love how we're demonstrating that we can get the error if the request to delete fails, but I almost feel like this is too much of a behavior we want to avoid to include it in the official docs. The attached screenshot tells the story enough on its own I think: there's no useful or actionable information present in that message and it's impact will be spreading fear when nobody can do anything about it.

If we want to show error handling I have these questions:
 - what kind of errors should we expect to get? not conceptually, but what is the return value of `getLastEntityDeleteError`? not great that we recreated C-style "get last error" calling patterns, but if we do that, can we at least expect some error codes or stable strings?
 - how would we ideally communicate the error to the person at the screen? this doc has great examples of using spinners and indicators to communicate where in the process we are, but for errors we are kinda just dropping the ball and giving up. could we use a Toast message of some kind to indicate the failure?
 - how should we respond after failure? retry? disable some control? even a helpful string message as used on invalid form fields could provide more actionable information.

if these docs didn't have the power to establish patterns and guide people's habits I wouldn't consider this as big of a deal, but I think we should write these with the idea that whatever we put out there is what we are telling the community we want to see. it would be worth removing this section or investing a little more in it to set a good example.

Reply:
Good point! I followed the same pattern as the rest of WordPress and added Snackbar displaying the `error.message`.